managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryImpl.java

```diff
@@ -47,56 +47,61 @@ protected EntryImpl newObject(Handle<EntryImpl> handle) {
     private Runnable onDeallocate;
 
     public static EntryImpl create(LedgerEntry ledgerEntry) {
-        EntryImpl entry = RECYCLER.get();
+        EntryImpl entry = getEntryFromRecycler();
         entry.timestamp = System.nanoTime();
         entry.ledgerId = ledgerEntry.getLedgerId();
         entry.entryId = ledgerEntry.getEntryId();
         entry.data = ledgerEntry.getEntryBuffer();
         entry.data.retain();
-        entry.setRefCnt(1);
+        return entry;
+    }
+
+    private static EntryImpl getEntryFromRecycler() {
+        EntryImpl entry = RECYCLER.get();
+        if (entry.refCnt() != 1) {
+            throw new IllegalStateException(
+                    "EntryImpl should be obtained from the recycler with refCnt == 1. refCnt = " + entry.refCnt()
+                            + " instance = " + entry);
+        }
```

Comment:
I'm wondering in which case could `RECYCLER.get()` return a reference object whose refcount is not 1.

Reply:
Yes, this check didn't make sense. I removed it.